packages/java/parser-jvm-core/src/main/java/com/vaadin/fusion/parser/core/AbstractRelative.java

```diff
@@ -0,0 +1,50 @@
+package com.vaadin.fusion.parser.core;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+
+abstract class AbstractRelative<T, P extends Relative<?>>
+        implements Relative<P> {
+    protected final T origin;
+    protected final P parent;
+
+    public AbstractRelative(@Nonnull T origin, P parent) {
+        this.origin = Objects.requireNonNull(origin);
+        this.parent = parent;
+    }
+
+    @Override
+    public T get() {
+        return origin;
+    }
+
+    @Override
+    public Optional<P> getParent() {
+        return Optional.of(parent);
+    }
+
+    @Override
```

Comment:
Yes. Since `parent` exists only for tranking, I thought it is better to consider that element that has everything but parent equal to another object, then objects are equal. Is it not java-ish?

Reply:
ok, just to double-check if it's forgotten accidentally